boards/arm/blackpill_f401cc/blackpill_f401cc.yaml

```diff
@@ -0,0 +1,16 @@
+identifier: blackpill_f401cc
+name: WeAct Studio Black Pill V3.0
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+supported:
+  - counter
+  - spi
+  - usb
+  - i2c
+  - uart
+  - pwm
+  - adc
```

Comment:
Compilation issue comes from the fact that you mention adc is supported but the adc test does not support the board. 
Please update `tests/drivers/adc/adc_api/src/test_adc.c`